src/TestUtils.jl

```diff
@@ -0,0 +1,173 @@
+# This file contains code that was formerly part of Julia. License is MIT: https://julialang.org/license
+
+module TestUtils
+
+export test_fft_backend
+
+using AbstractFFTs
+using AbstractFFTs: Plan
+
+using LinearAlgebra
+using Test
+
+"""
+    test_fft_backend(array_constructor; test_real=true, test_inplace=true) 
+
+Run tests to verify correctness of FFT functions using a particular 
+backend plan implementation. The backend implementation is assumed to be loaded 
+prior to calling this function.
+
+# Arguments
+
+- `array_constructor`: determines the `AbstractArray` implementation for
+  which the correctness tests are run. It is assumed to be a callable object that
+  takes in input arrays of type `Array` and return arrays of the desired type for
+  testing. For example, this can be a constructor such as `Array` or `CUDA.CuArray`.
+- `test_real=true`: whether to test real-to-complex and complex-to-real FFTs.
+- `test_inplace=true`: whether to test in-place plans. 
+"""
+function test_fft_backend(array_constructor; test_real=true, test_inplace=true)
```

Comment:
Another general comment: Could this be split into separate tests for `fft`, `bfft`, etc. so that one could test them separately?

Reply:
In theory yes, but it would require some refactoring work, as well as some thinking about what level of granularity we want to offer.

If you're okay with it, I would rather not do it in this PR? Although I've already made some improvements to coverage, this test set is essentially based on what was originally in `runtests.jl`.